addons/redis/scripts/redis-sentinel-start-v2.sh

```diff
@@ -42,6 +42,124 @@ build_redis_sentinel_conf() {
   echo "build redis sentinel conf succeeded!"
 }
 
+recover_registered_redis_servers_if_needed() {
+  if [ -f /data/sentinel/init_done.conf ]; then
+    echo "normal start"
+  else
+    echo "horizontal scaling"
+    if recover_registered_redis_servers; then
+      touch /data/sentinel/init_done.conf
+    else
+      echo "recover_registered_redis_servers failed"
+      exit 1
+    fi
+  fi
+}
+
+recover_registered_redis_servers() {
+    if [[ -n "${SENTINEL_POD_FQDN_LIST}" ]]; then
+        old_ifs="$IFS"
+        IFS=','
+        set -f
+        read -ra sentinel_pod_fqdn_list <<< "${SENTINEL_POD_FQDN_LIST}"
+        set +f
+        IFS="$old_ifs"
+
+        output=""
+        for sentinel_pod_fqdn in "${sentinel_pod_fqdn_list[@]}"; do
+          if [ -n "$SENTINEL_PASSWORD" ]; then
+            temp_output=$(redis-cli -h "$sentinel_pod_fqdn" -p 26379 -a "$SENTINEL_PASSWORD" sentinel masters 2>/dev/null || true)
+          else
+            temp_output=$(redis-cli -h "$sentinel_pod_fqdn" -p 26379 sentinel masters 2>/dev/null || true)
+          fi
+          if [[ -n "$temp_output" ]]; then
+              output="$temp_output"
+              break
```

Comment:
Better to check if the masters configuration of all Sentinel instances is consistent. If there are inconsistencies, fail immediately. It is necessary to first ensure the consistency of the configuration across other nodes.